firmware/library/L2_Utilities/utility.cpp

```diff
@@ -0,0 +1,94 @@
+#include <cstdlib>
+
+#include "utility.hpp"
+
+uint32_t utility::Random(uint32_t min, uint32_t max)
+{
+    return rand() % (max - min + 1) + min;
+}
+void utility::SetRandomSeed(uint32_t seed)
+{
+    srand(seed);
+}
+uint32_t utility::BitExtract(uint32_t target, uint8_t start_bit, uint8_t width)
+{
+    uint32_t mask = 0;
+    for (int i = 0; i < width; i++)
+    {
+        mask = (mask << 1) + 1;
+    }
+    uint8_t true_start_bit = static_cast<uint8_t>(start_bit - (width - 1));
+    mask <<= true_start_bit;
+    return (target & mask) >> true_start_bit;
+}
+uint32_t utility::BitPlace(uint32_t target, uint8_t start_bit,
+                           uint32_t new_bits, uint8_t width)
+{
+    uint32_t temp = target;
+    for (uint8_t i = 0; i < width; i++)
+    {
+        temp = utility::BitClear(target, uint8_t(start_bit - i));
+    }
+    uint32_t mask = new_bits;
+    mask <<= (start_bit - (width - 1));
+    return temp | mask;
+}
+
+uint32_t utility::BitSet(uint32_t target, uint8_t bit_position)
```

Comment:
Would be useful to put a `bit_width` parameter for this that defaults to 1. So the user can set multiple bits.